### ISSUE TITLE
Incorrect file name mangling with dual extensions files

### ISSUE THREAD

**Issue by thangngoc89** [pocketbase/pocketbase#4824]
The file name is `foo.nii.gz` and pocketbase turn it to `foo_nii_0123456789.gz` with 0123456789 is the random bits.
Even though this is technically correct, `.nii.gz` is the defacto format for working with NIfTI files.

So I'm asking for a way to whitelist `.nii.gz` or allowing to change the download filename somehow because most software would expect `.nii.gz` extension

**Comment by ganigeorgiev** [comment]
I'll investigate it more details tomorrow and eventually will consider preserving double extensions.

**Comment by thangngoc89** [comment]
@ganigeorgiev thank you for your quick response. In the meantime, I've found the code snippet for changing the filename on the fly using hook in https://github.com/pocketbase/pocketbase/issues/306#issuecomment-1214146730 

I'm trying to translate it to Javascript without success.

This is what I had so far:

```js
onFileDownloadRequest((e) => {
  fs = $app.newFilesystem();
  fs.serve(e.httpContext.response(), e.servedPath, "test.zip");
  fs.close();

  return __hooks.StopPropagation;
});
```

Any tips?

**Comment by ganigeorgiev** [comment]
I've pushed a fix in `master` to take double extensions in consideration and it will be available with the next minor v0.22.10 release (most likely tomorrow or the day following it).

Regarding the hook, `__hooks` is a string and doesn't have `StopPropagation` field. To stop the hook chain in the JSVM you need to either throw an error OR return `false` (_note that the hooks will slightly change with the [refactoring](https://github.com/pocketbase/pocketbase/discussions/4355)_).
